downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/MapOldStableIds.java

```diff
@@ -0,0 +1,123 @@
+package org.reactome.release.downloadDirectory;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.ReactomeJavaConstants;
+
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class MapOldStableIds {
+	private static final Logger logger = LogManager.getLogger();
+	
+	private static Connection connect = null;
+	private static Statement statement = null;
+	private PreparedStatement preparedStatement = null;
+	private static ResultSet resultSet = null;
+	
+	public static void execute(String username, String password, String host, String releaseNumber) throws Exception {
+		logger.info("Running MapOldStableIds");
+		// Need to use mysql driver to access stable_identifiers db
+		logger.info("Connecting to stable_identifers db...");
+		Class.forName("com.mysql.jdbc.Driver");
+		connect = DriverManager.getConnection("jdbc:mysql://" + host + "/stable_identifiers?" + "user=" + username + "&password=" + password);
+		statement = connect.createStatement();
+		resultSet = statement.executeQuery("SELECT identifier,instanceId FROM StableIdentifier");
+		
+		// Iterate through returned results of DB IDs and stable IDs 
+		logger.info("Mapping Old Stable IDs to Current Stable IDs...");
+		HashMap<String,ArrayList<String>> dbIdToStableIds = new HashMap<String,ArrayList<String>>();
+		ArrayList<String> dbIds = new ArrayList<String>();
+		while (resultSet.next()) {
+			String stableId = resultSet.getString(1);
+			String dbId = resultSet.getString(2);
+			
+			if (dbIdToStableIds.get(dbId) != null) {
+				dbIdToStableIds.get(dbId).add(stableId);
+			} else {
+				ArrayList<String> stableIds = new ArrayList<String>(Arrays.asList(stableId));
+				dbIdToStableIds.put(dbId, stableIds);
```

Comment:
The stableIds temp could be removed using this `dbIdToStableIds.put(dbId, Collections.singletonList(stableId));`

Reply:
The while loop could actually be extracted to a method:
```
Map<String, List<String>> dbIdToStableIds = getDbIdToStableIds(resultSet);
List<String> dbIds = new ArrayList<>(dbIdToStableIds.keySet());
Collections.sort(dbIds);


private static Map<String, List<String>> getDbIdToStableIds(ResultSet resultSet) throws SQLException {
    Map<String, List<String>> dbIdToStableIds = new HashMap<>();

    // Iterate through returned results of DB IDs and stable IDs
    while (resultSet.next()) {
        String stableId = resultSet.getString(1);
        String dbId = resultSet.getString(2);

        if (dbIdToStableIds.get(dbId) != null) {
            dbIdToStableIds.get(dbId).add(stableId);
        } else {
            List<String> stableIds = new ArrayList<>(Collections.singletonList(stableId));
            dbIdToStableIds.put(dbId, stableIds);
        }
    }

    return dbIdToStableIds;
}
```